### DISCUSSION TITLE
I've git committed and pushed my latest changes, but when I try and redeploy, I get the old errors, why?

### DISCUSSION THREAD

**Discussion by DomSpicer** [vercel/community#3571] [Help] [answered]
### Summary

So as the title shows, I had a few problems with my deployment, so I corrected those errors, committed the changes and pushed them to Git. I then went to redeploy, but I got the same old errors, despite correcting those errors. Any ideas why?

Initially, I had set up some routing for a navbar which I've never really done before (I'm learning), but I was getting a 404 error. Which I still am, but then I had issues with deployment which notified me of other issues...

### Steps to Reproduce

These are the errors, which have been fixed.

[07:51:22.480] Running build in San Francisco, USA (West) – sfo1
[07:51:22.541] Cloning github.com/DomSpicer/photoportfolio (Branch: main, Commit: 4d98464)
[07:51:22.547] Skipping build cache, deployment was triggered without cache.
[07:51:23.506] Cloning completed: 965.088ms
[07:51:23.726] Running "vercel build"
[07:51:24.229] Vercel CLI 31.2.3
[07:51:24.478] Installing dependencies...
[07:51:37.116] 
[07:51:37.116] > photoportfolio@0.1.0 postinstall
[07:51:37.116] > prisma generate
[07:51:37.116] 
[07:51:37.738] Prisma schema loaded from prisma/schema.prisma
[07:51:38.199] 
[07:51:38.200] ✔ Generated Prisma Client (5.1.0 | library) to ./node_modules/@prisma/client in 87ms
[07:51:38.200] You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
[07:51:38.200] ```
[07:51:38.200] import { PrismaClient } from '@prisma/client'
[07:51:38.200] const prisma = new PrismaClient()
[07:51:38.200] ```
[07:51:38.237] 
[07:51:38.237] added 451 packages in 13s
[07:51:38.237] 
[07:51:38.237] 153 packages are looking for funding
[07:51:38.238]   run `npm fund` for details
[07:51:38.270] Detected Next.js version: 13.4.12
[07:51:38.273] Detected `package-lock.json` generated by npm 7+...
[07:51:38.273] Running "npm run build"
[07:51:38.575] 
[07:51:38.576] > photoportfolio@0.1.0 build
[07:51:38.576] > next build
[07:51:38.576] 
[07:51:39.088] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[07:51:39.089] This information is used to shape Next.js' roadmap and prioritize features.
[07:51:39.089] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[07:51:39.089] https://nextjs.org/telemetry
[07:51:39.089] 
[07:51:39.179] - info Linting and checking validity of types...
[07:51:45.238] 
[07:51:45.238] Failed to compile.
[07:51:45.239] 
[07:51:45.239] ./src/pages/main/index.tsx
[07:51:45.239] 3:8  Warning: 'Link' is defined but never used.  @typescript-eslint/no-unused-vars
[07:51:45.239] 6:10  Warning: 'global' is defined but never used.  @typescript-eslint/no-unused-vars
[07:51:45.239] 11:11  Warning: 'data' is assigned a value but never used.  @typescript-eslint/no-unused-vars
[07:51:45.239] 25:54  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
[07:51:45.239] 
[07:51:45.239] ./src/server/api/routers/MainRoutes.tsx
[07:51:45.239] 9:25  Warning: 'useLocation' is defined but never used.  @typescript-eslint/no-unused-vars
[07:51:45.240] 
[07:51:45.240] ./src/server/api/routers/posts.ts
[07:51:45.240] 1:10  Warning: 'z' is defined but never used.  @typescript-eslint/no-unused-vars
[07:51:45.240] 
[07:51:45.240] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[07:51:45.262] Error: Command "npm run build" exited with 1

**Note, I was able to run npm run build on vscode and it worked fine, but not on cmd.**

See below for the code: https://vscode.dev/github/DomSpicer/photoportfolio/blob/main

After I was able to git commit and push, whilst being able to run: npm run build directly on vscode and for it to work fine... I then get these errors with a different commit:

<html>
<body>

ReferenceError: document is not defined
--
07:55:50.052 | at getUrlBasedHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:398:14)
07:55:50.052 | at Object.createBrowserHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:231:10)
07:55:50.052 | at e.BrowserRouter (/vercel/path0/node_modules/react-router-dom/dist/umd/react-router-dom.production.min.js:11:15009)
07:55:50.053 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.053 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.053 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.053 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
07:55:50.053 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.054 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
07:55:50.054 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.054 |  
07:55:50.054 | Error occurred prerendering page "/main/_app". Read more: https://nextjs.org/docs/messages/prerender-error
07:55:50.054 | ReferenceError: document is not defined
07:55:50.054 | at getUrlBasedHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:398:14)
07:55:50.054 | at Object.createBrowserHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:231:10)
07:55:50.055 | at e.BrowserRouter (/vercel/path0/node_modules/react-router-dom/dist/umd/react-router-dom.production.min.js:11:15009)
07:55:50.057 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.057 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.057 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.057 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
07:55:50.057 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.058 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
07:55:50.058 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.058 | - info Generating static pages (4/18)
07:55:50.081 | - info Generating static pages (8/18)
07:55:50.113 | ReferenceError: document is not defined
07:55:50.113 | at getUrlBasedHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:398:14)
07:55:50.113 | at Object.createBrowserHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:231:10)
07:55:50.113 | at e.BrowserRouter (/vercel/path0/node_modules/react-router-dom/dist/umd/react-router-dom.production.min.js:11:15009)
07:55:50.113 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.113 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.113 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.113 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
07:55:50.113 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.114 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
07:55:50.114 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.114 |  
07:55:50.114 | Error occurred prerendering page "/en/main/_app". Read more: https://nextjs.org/docs/messages/prerender-error
07:55:50.114 | ReferenceError: document is not defined
07:55:50.114 | at getUrlBasedHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:398:14)
07:55:50.114 | at Object.createBrowserHistory (/vercel/path0/node_modules/@remix-run/router/dist/router.cjs.js:231:10)
07:55:50.115 | at e.BrowserRouter (/vercel/path0/node_modules/react-router-dom/dist/umd/react-router-dom.production.min.js:11:15009)
07:55:50.115 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.115 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.115 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.119 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
07:55:50.119 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.119 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
07:55:50.120 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.120 | - info Generating static pages (13/18)
07:55:50.338 | Error: UserContext not found
07:55:50.338 | at assertContextExists (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:38:42)
07:55:50.338 | at useCtx (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:47:5)
07:55:50.339 | at useUser (/vercel/path0/node_modules/@clerk/clerk-react/dist/cjs/hooks/useUser.js:26:54)
07:55:50.339 | at Home (/vercel/path0/.next/server/pages/main.js:106:72)
07:55:50.339 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.339 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.339 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.339 | at Xc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:409)
07:55:50.339 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:210)
07:55:50.339 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.339 |  
07:55:50.339 | Error occurred prerendering page "/main". Read more: https://nextjs.org/docs/messages/prerender-error
07:55:50.339 | Error: UserContext not found
07:55:50.340 | at assertContextExists (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:38:42)
07:55:50.340 | at useCtx (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:47:5)
07:55:50.340 | at useUser (/vercel/path0/node_modules/@clerk/clerk-react/dist/cjs/hooks/useUser.js:26:54)
07:55:50.340 | at Home (/vercel/path0/.next/server/pages/main.js:106:72)
07:55:50.340 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.340 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.340 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.340 | at Xc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:409)
07:55:50.340 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:210)
07:55:50.340 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.370 | Error: UserContext not found
07:55:50.371 | at assertContextExists (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:38:42)
07:55:50.371 | at useCtx (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:47:5)
07:55:50.371 | at useUser (/vercel/path0/node_modules/@clerk/clerk-react/dist/cjs/hooks/useUser.js:26:54)
07:55:50.371 | at Home (/vercel/path0/.next/server/pages/main.js:106:72)
07:55:50.371 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.371 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.371 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.371 | at Xc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:409)
07:55:50.371 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:210)
07:55:50.371 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.372 |  
07:55:50.372 | Error occurred prerendering page "/en/main". Read more: https://nextjs.org/docs/messages/prerender-error
07:55:50.372 | Error: UserContext not found
07:55:50.372 | at assertContextExists (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:38:42)
07:55:50.372 | at useCtx (/vercel/path0/node_modules/@clerk/shared/dist/cjs/hooks/createContextAndHook.js:47:5)
07:55:50.372 | at useUser (/vercel/path0/node_modules/@clerk/clerk-react/dist/cjs/hooks/useUser.js:26:54)
07:55:50.372 | at Home (/vercel/path0/.next/server/pages/main.js:106:72)
07:55:50.372 | at Wc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
07:55:50.372 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
07:55:50.372 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.372 | at Xc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:409)
07:55:50.372 | at Zc (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:210)
07:55:50.372 | at Z (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
07:55:50.373 | - info Generating static pages (18/18)
07:55:50.378 |  
07:55:50.379 | > Export encountered errors on following paths:
07:55:50.379 | /main
07:55:50.380 | /main/_app
07:55:50.380 | /main/_app: /en/main/_app
07:55:50.380 | /main: /en/main
07:55:50.421 | Error: Command "npm run build" exited with 1

</body>
</html>

As I said I'm learning, so any tips would really help :)

**Comment by stormyzio** [comment]
Hello! Maybe run "npm install" to install all dependencies. I'm not sure but it's maybe a solution.

**Reply by DomSpicer** [reply]
Hi! Thanks for the reply, unfortunately it hasn't done anything, the errors still persist.

**Comment by amyegan** [comment, marked as answer]
Hey @DomSpicer. If you **redeploy** from an existing deployment, it's going to try to rebuild from the same commit as the original. You should [create a new deployment](https://vercel.com/docs/concepts/deployments/overview) to include your latest code changes.

**Reply by DomSpicer** [reply]
That's done it, thanks Amy!
